### PR TITLE
Project setup and starter code

### PR DESCRIPTION
**Pull request by tkz96**
This PR is to:

- Add skeleton code
- Set project up with necessary files

### CODE REVIEW

File: school_person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
```
Comment (Deepakdanger) [inline school_person.rb]: - [ ] Kindly move this function "can_use_services?" above the "private" line 12.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @tkz96, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: school_student.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(classroom)
+    super
+    @classroom = classroom
+  end
```
Comment (thecodechaser) [inline school_student.rb:5]: - [ ] [optional] The Student initializer extends the parent's initializer and should also accept parameters that are required for the parent class person. I would suggest considering the below code.
  - `def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end`

File: school_teacher.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(specialization)
+    super
+    @specialization = specialization
+  end
```
Comment (thecodechaser) [inline school_teacher.rb:5]: - [ ] [optional] The Teacher initializer extends the parent's initializer and should also accept parameters that are required for the parent class Person. I would suggest considering the below code.
  - `def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end`

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @tkz96👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Optional suggestions

- [ ] I'm marking these changes as optional because they were missed by the previous reviewer but I strongly suggest considering them.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: school_teacher.rb (hunk 3 above)
Comment (tkz96) [reply school_teacher.rb:5]: Thank you for your suggestions. I will consider them if needed.

**Pull request merged**